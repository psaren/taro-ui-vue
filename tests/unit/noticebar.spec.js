import { mount } from '@vue/test-utils'
import AtNoticebar from '../../src/components/noticebar'

const factory = (values = {}, slots = { default: ['这是内容'] }) => {
  return mount(AtNoticebar, {
    slots,
    propsData: { ...values },
  })
}

describe('AtNoticebar Snap', () => {
  it('render AtNoticebar -- props show', () => {
    const wrapper = factory({ close: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNoticebar -- props single', () => {
    const wrapper = factory({ single: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNoticebar -- props speed', () => {
    const wrapper = factory({ speed: 200 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNoticebar -- props moreText & showMore', () => {
    // showMore work only when single is true
    const wrapper = factory({ moreText: '查看更多', showMore: true, single: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtNoticebar -- props icon', () => {
    const wrapper = factory({ icon: 'volume-plus' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtNoticebar Event', () => {
  it('AtNoticebar onClose', () => {
    const onClose = jest.fn()
    const wrapper = factory({
      close: true,
      onClose: onClose,
    })
    wrapper.find('.at-noticebar__close').trigger('tap')
    expect(onClose).toBeCalled()
  })

  it('AtNoticebar onGotoMore', () => {
    const onGotoMore = jest.fn()
    const wrapper = factory({
      icon: 'volume-plus',
      single: true,
      showMore: true,
      onGotoMore: onGotoMore,
      moreText: '更多内容',
    })
    wrapper.find('.at-noticebar__more').trigger('tap')
    expect(onGotoMore).toBeCalled()
  })
})
