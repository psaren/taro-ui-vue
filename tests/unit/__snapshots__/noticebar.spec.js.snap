// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtNoticebar Snap render AtNoticebar -- props icon 1`] = `
<view
  class="at-noticebar"
>
  <view
    class="at-noticebar__content"
  >
    <view
      class="at-noticebar__content-icon"
    >
      <view
        class="at-icon at-icon-volume-plus at-icon at-icon-volume-plus"
      />
    </view>
    <view
      class="at-noticebar__content-text"
    >
      <view
        animation="[object Object]"
        class="at-noticebar__content-inner"
      >
        这是内容
      </view>
    </view>
  </view>
</view>
`;

exports[`AtNoticebar Snap render AtNoticebar -- props moreText & showMore 1`] = `
<view
  class="at-noticebar at-noticebar--single"
>
  <view
    class="at-noticebar__content"
  >
    <view
      class="at-noticebar__content-text"
    >
      <view
        animation="[object Object]"
        class="at-noticebar__content-inner"
      >
        这是内容
      </view>
    </view>
  </view>
  <view
    class="at-noticebar__more"
  >
    <view
      class="text"
    >
      查看更多
    </view>
    <view
      class="at-noticebar__more-icon"
    >
      <view
        class="at-icon at-icon-chevron-right"
      />
    </view>
  </view>
</view>
`;

exports[`AtNoticebar Snap render AtNoticebar -- props show 1`] = `
<view
  class="at-noticebar"
>
  <view
    class="at-noticebar__close"
  >
    <view
      class="at-icon at-icon-close"
    />
  </view>
  <view
    class="at-noticebar__content"
  >
    <view
      class="at-noticebar__content-text"
    >
      <view
        animation="[object Object]"
        class="at-noticebar__content-inner"
      >
        这是内容
      </view>
    </view>
  </view>
</view>
`;

exports[`AtNoticebar Snap render AtNoticebar -- props single 1`] = `
<view
  class="at-noticebar at-noticebar--single"
>
  <view
    class="at-noticebar__content"
  >
    <view
      class="at-noticebar__content-text"
    >
      <view
        animation="[object Object]"
        class="at-noticebar__content-inner"
      >
        这是内容
      </view>
    </view>
  </view>
</view>
`;

exports[`AtNoticebar Snap render AtNoticebar -- props speed 1`] = `
<view
  class="at-noticebar"
>
  <view
    class="at-noticebar__content"
  >
    <view
      class="at-noticebar__content-text"
    >
      <view
        animation="[object Object]"
        class="at-noticebar__content-inner"
      >
        这是内容
      </view>
    </view>
  </view>
</view>
`;
